src/test/java/seam/EnergyCalculatorTest.java

```diff
@@ -0,0 +1,173 @@
+package seam;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.awt.*;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class EnergyCalculatorTest
+{
+    private Pixel[][] pixels;
+    private int imageWidth;
+    private int imageHeight;
+    private final int max = 255 * 255 * 6;
+
+    @BeforeEach
+    public void setUpValues()
+    {
+        imageWidth = 3;
+        imageHeight = 4;
+        pixels = new Pixel[imageWidth][imageHeight];
+        Color[] colors = new Color[]{
+                Color.RED, Color.ORANGE, Color.YELLOW, Color.GREEN,
+                Color.BLUE, Color.PINK, Color.BLACK, Color.WHITE,
+                Color.LIGHT_GRAY, Color.DARK_GRAY, Color.MAGENTA, Color.CYAN
+        };
+
+        int index = 0;
+        for (int i = 0; i < pixels.length; i++)
+        {
+            for (int j = 0; j < pixels[i].length; j++)
+            {
+                pixels[i][j] = new Pixel(colors[index++].getRGB());
+            }
+        }
+
+    }
+
+    @Test
+    public void calculateEnergyTest()
+    {
+        // given
+        EnergyCalculator energyCalculator
+                = new EnergyCalculator(pixels, imageWidth, imageHeight);
+
+        // when
+        energyCalculator.calculateEnergy();
+
+        // then
+
+        // some border pixels
+        assertEquals(max, pixels[0][0].getEnergy());
+        assertEquals(max, pixels[0][imageHeight - 1].getEnergy());
+        assertEquals(max, pixels[imageWidth - 1][0].getEnergy());
+
+        // middle pixel: pixels[1][1]
+        Color top = pixels[0][1].getColor();
+        Color bottom = pixels[2][1].getColor();
+        Color left = pixels[1][0].getColor();
+        Color right = pixels[1][2].getColor();
+        double energy = ((top.getRed() - bottom.getRed())
+                * (top.getRed() - bottom.getRed()))
+                + ((top.getGreen() - bottom.getGreen())
+                * (top.getGreen() - bottom.getGreen()))
+                + ((top.getBlue() - bottom.getBlue())
+                * (top.getBlue() - bottom.getBlue()))
+                + ((left.getRed() - right.getRed())
+                * (left.getRed() - right.getRed()))
+                + ((left.getGreen() - right.getGreen())
+                * (left.getGreen() - right.getGreen()))
+                + ((left.getBlue() - right.getBlue())
+                * (left.getBlue() - right.getBlue()));
```

Comment:
I would replace this with a value you calculate on paper, rather than copy/paste the energy code here.